js/src/helper.ts

```diff
@@ -18,9 +18,9 @@ export function serializePath(path: string): Buffer {
     let value = 0
     let child = pathArray[i]
     if (child.endsWith("'")) {
-      value += HARDENED
       child = child.slice(0, -1)
     }
+    value += HARDENED
```

Comment:
no this won't do at all! it would be wrong to harden every single element... it should be hardened only when requested